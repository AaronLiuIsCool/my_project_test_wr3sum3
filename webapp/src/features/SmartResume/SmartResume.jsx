import React from 'react';

import LeftNav from './components/LeftNav';
import ExperiencesForm from './components/ExperiencesForm';
import Preview from './components/Preview';

const SmartResume = () => (
    <div>
        <div>智能创建</div>
        <LeftNav />
        <ExperiencesForm />
        <Preview />
    </div>
);

export default SmartResume;