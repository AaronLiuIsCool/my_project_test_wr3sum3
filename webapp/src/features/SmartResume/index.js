import React, { Suspense } from 'react';

import LoadFallbackComponent from 'components/LoadFallbackComponent';
import './styles/index.scss';

const SmartResume = React.lazy(() => import('./components/SmartResume'));

const LazyLoadSmartResume = ({ match }) => (
  <Suspense fallback={<LoadFallbackComponent />}>
    <SmartResume useObserver={true} resumeId={match.params.resumeId} />
  </Suspense>
);

export default LazyLoadSmartResume;
