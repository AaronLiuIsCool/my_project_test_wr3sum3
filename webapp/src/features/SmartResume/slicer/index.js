import { createSlice } from "@reduxjs/toolkit";

import assistant from './assistant';
import photoReference from "./photoReference";
import basic, {validateBasic} from "./basic";
import education, {validateEducation} from "./education";
import work, {validateWork} from "./work";
import project, {validateProject} from "./project";
import volunteer, {validateVolunteer} from "./volunteer";
import certificate, {validateCertificate} from "./certificate";
import ratings from "./ratings";
import resumeBuilder from "./resumeBuilder";
import { resumeAdaptor } from "../utils/slicerAdaptor";

export const resumeSlice = createSlice({
    name: "resume",
    initialState: {
        id: undefined,
        stepIndex: 0,

        assistant: assistant.initialState,
        photoReference: photoReference.initialState,
        basic: basic.initialState,
        education: education.initialState,
        work: work.initialState,
        project: project.initialState,
        certificate: certificate.initialState,
        volunteer: volunteer.initialState,
        ratings: ratings.initialState,
        resumeBuilder: resumeBuilder.initialState,
    },
    reducers: {
        setId: (state, action) => {
            state.id = action.payload;
        },

        setResume: (state, action) => {
            const resume = action.payload;
            return { ...state, ...resumeAdaptor(resume) };
        },

        moveStep: (state, action) => {
            state.stepIndex = action.payload;
        },

        ...assistant.reducers,
        ...photoReference.reducers,
        ...basic.reducers,
        ...education.reducers,
        ...work.reducers,
        ...project.reducers,
        ...volunteer.reducers,
        ...certificate.reducers,
        ...ratings.reducers,
        ...resumeBuilder.reducers,
    }
});

export const { actions } = resumeSlice;

export const selectId = ({ resume }) => resume.id;
export const selectStep = ({ resume }) => resume.stepIndex;
export const selectResume = ({ resume }) => resume;
export const selectValidatedResume = ({ resume }) => {
  const {photoReference, basic, education, work,
    project, volunteer, certificate} = resume;

  return {
    photoReference,
    basic: validateBasic(basic) ? basic : undefined,
    education: {...education, data: education.data.filter(validateEducation)},
    work: {...work, data: work.data.filter(validateWork)},
    project: {...project, data: project.data.filter(validateProject)},
    volunteer: {...volunteer, data: volunteer.data.filter(validateVolunteer)},
    certificate: {...certificate, data: certificate.data.filter(validateCertificate)},
  };
};

export const assistantSelectors = assistant.selectors;
export const photoReferenceSelectors = photoReference.selectors;
export const basicSelectors = basic.selectors;
export const educationSelectors = education.selectors;
export const workSelectors = work.selectors;
export const projectSelectors = project.selectors;
export const volunteerSelectors = volunteer.selectors;
export const certificateSelectors = certificate.selectors;
export const ratingsSelectors = ratings.selectors;
export const resumeBuilderSelectors = resumeBuilder.selectors;

export default resumeSlice.reducer;
