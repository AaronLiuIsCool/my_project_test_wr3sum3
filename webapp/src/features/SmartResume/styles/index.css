/* This file contains some shared css that can be used cross SmartResume */

.form_h1{
  text-align:left;
  margin: 40px 0;
}

.form_h1 h1{
  font-size: 30px;
  line-height: 1;
  color: #333333;
}

.form_h1 p{
  font-size: 14px;
  line-height: 2.14;
  color: #bfc4d3;
}

.form_h2{
  font-size: 20px;
  font-weight: bold;
  color: #333333;
}

.form_body{
  width: 800px;
  border-radius: 8px;
  box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.1);
  padding: 24px 20px;
}

.form_item label{
  color: #bfc4d3;
}
.form_item .rbt-input-main,
.form_item input{
  background-color: #f2f4f9;
  color: #d5d9e6;
  border-radius: 8px;
  border:none;
  /* box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.1); */
}
.space_betweens{
  display: flex;
  justify-content: space-between;
}
.form_reset_link{
  font-size: 16px;
  font-weight: bold;
  color: #d5d9e6;
}


/* 添加其他院校/工作/专业项目经历 按钮 */
.addMoreForm{
  display:flex;
  padding:20px;
  color: #3e89ec;
}
/* if svg is provided, we don't need this  */
.addMoreForm span{
  font-size: 50px;
  line-height: 20px;
}
.addMoreForm p{
  font-size: 20px;
  font-weight: bold;
  padding-left:10px;
}