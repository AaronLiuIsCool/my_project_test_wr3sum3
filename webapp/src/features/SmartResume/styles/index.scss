/* This file contains some shared css that can be used cross SmartResume */
.smart-resume {
  position: relative;

  a:hover {
    text-decoration: none;
  }

  .overlay {
    position: fixed;
    width: 100%;
  }

  .form_h1 {
    text-align: left;
    margin: 40px 0;

    div {
      display: flex;
      align-items: center;

      h1 {
        font-size: 30px;
        line-height: 1;
        color: #333333;
        margin-left: 10px;
        // icon has top margin, so we need manually adjust this 
        margin-top: 5px;
      }
    }

    p {
      font-size: 14px;
      line-height: 2.14;
      color: #bfc4d3;
    }
  }

  .form_h2 {
    font-size: 20px;
    font-weight: bold;
    color: #333333;
    margin-bottom: 10px;
  }

  .form_body {
    width: 800px;
    border-radius: 8px;
    box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.1);
    padding: 24px 20px;
  }

  .form_item {
    label {
      color: #bfc4d3;
    }

    .rbt-input-main,
    textarea,
    input {
      background-color: #f2f4f9;
      color: #333333;
      border-radius: 8px;
      border: none;
      box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.1);
    }

    .DateInput_input {
      height: 38px;
      font-size: 16px;
      line-height: 24px;

      &::placeholder {
        color: black;
      }
    }

    .radioButton_container {
      display: flex;
      justify-content: space-around;

      .radioButton {
        width: 38px;
        height: 38px;
        border-radius: 21px;
        border: none;
        background: white;
        color: #bfc4d3;

        &.selected {
          color: white;
          background-color: #3e89ec;
        }

        &:focus {
          border: none;
          outline: none;
        }
      }
    }


  }

  .form-group textarea {
    background-color: #f2f4f9;
    color: #d5d9e6;
    border-radius: 8px;
    border: none;
    color: #333;
  }

  .form_item .rbt-input-main,
  .form_item input {
    background-color: #f2f4f9;
    color: #d5d9e6;
    border-radius: 8px;
    border: none;
    color: #333;
  }

  .space_betweens {
    display: flex;
    justify-content: space-between;
  }

  .form_reset_link {
    font-size: 16px;
    font-weight: bold;
    color: #d5d9e6;
  }

  .w-100 {
    width: 100%;
    border-radius: 30px;
  }

  .SingleDatePicker {
    display: block;

    .SingleDatePickerInput {
      width: 100%;

      .DateInput {
        width: 100%;
      }
    }
  }


  /* 添加其他院校/工作/专业项目经历 按钮 */
  .addMoreForm {
    display: flex;
    padding: 20px;
    color: #3e89ec;
    cursor:pointer;

    /* if svg is provided, we don't need this  */
    span {
      font-size: 50px;
      line-height: 20px;
    }

    p {
      font-size: 20px;
      font-weight: bold;
      padding-left: 10px;
    }
  }

  .writeAssistantContainer {
    position: relative;

    .writeAssistant {
      position: absolute;
      top: 0;
      right: 0;
      cursor: pointer;

      .hand {
        fill: #bfc4d3;
      }

      .btn.btn-link {
        padding-left: 10px;
        font-size: 14px;
        font-weight: bold;
        color: #bfc4d3;
      }
    }

    &.active, &:hover {
      .hand {
        fill: #3e89ec;
      }

      .btn.btn-link {
        text-decoration: none;
        color: #3e89ec;
      }
    }
  }

  // input placeholder text color 
  ::-webkit-input-placeholder {
    /* Chrome/Opera/Safari */
    color: #d5d9e6 !important;
  }

  ::-moz-placeholder {
    /* Firefox 19+ */
    color: #d5d9e6 !important;
  }

  :-ms-input-placeholder {
    /* IE 10+ */
    color: #d5d9e6 !important;
  }

  :-moz-placeholder {
    /* Firefox 18- */
    color: #d5d9e6 !important;
  }

  // overwrite some date picker style 
  .CalendarDay__selected,
  .CalendarDay__selected:active,
  .CalendarDay__selected:hover {
    background: #3e89ec !important;
    border-color: #3e89ec !important;
  }

  .DayPickerKeyboardShortcuts_show__bottomRight::before {
    border-top: 26px solid transparent !important;
    border-right: 33px solid #3e89ec !important;
  }

  /* resume preview style  */
  .react-pdf__Page canvas{
    margin:20px auto;
  }
}