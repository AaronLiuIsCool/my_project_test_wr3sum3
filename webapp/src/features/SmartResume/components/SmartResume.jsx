import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { previewResume } from './ResumePreview/resumeBuilder';

import { selectLanguage, selectUserId } from 'features/App/slicer';
import { actions } from 'features/SmartResume/slicer';
import { I8nContext } from 'shell/i18n';
import AccountServices from 'shell/services/AccountServices';
import ResumeServices from 'shell/services/ResumeServices';
import { getLogger } from 'shell/logger';

import LeftNav from './LeftNav';
import ExperiencesForm from './ExperiencesForm';
import Assistant from './Assistant';
import ResumePreview from './ResumePreview';

import zh from '../i18n/zh.json';
import en from '../i18n/en.json';

const logger = getLogger('ResumeStarter');
const accountServices = new AccountServices();
const resumeServices = new ResumeServices();

async function getOrCreateResume(dispatch, userId, resumeId, language) {
    if (resumeId) {
        try {
            const response = await resumeServices.getResume(resumeId);
            const resumeData = await response.json();
            dispatch(actions.setResume(resumeData));
        } catch (exception) {
            logger.error(exception)
        }
        return;
    }

    try {
        const response = await resumeServices.createResume({language});
        const data = await response.json();
        const resumeId = data.id;
        accountServices.addResume(userId, resumeId);
        dispatch(actions.setId(resumeId));
    } catch (exception) {
        logger.error(exception);
        // TODO: Need to handle retry
    }
}

const SmartResume = ({ useObserver = false, resumeId }) => {
    const dispatch = useDispatch();
    const userId = useSelector(selectUserId);
    const language = useSelector(selectLanguage);
    const messages = language === 'zh' ? zh : en;

    useEffect(() => {
        getOrCreateResume(dispatch, userId, resumeId, language);
		previewResume(messages.RPreview);
    }, []); // eslint-disable-line

    return (
        <I8nContext.Provider value={messages}>
            <div className="features smart-resume">
                <div className="overlay">
                    <LeftNav />
                    <Assistant />
                    <ResumePreview  />
                </div>
                <ExperiencesForm useObserver={useObserver} />
            </div>
        </I8nContext.Provider>
    );
};

SmartResume.propTypes = {
    useObserver: PropTypes.bool 
};

export default SmartResume;