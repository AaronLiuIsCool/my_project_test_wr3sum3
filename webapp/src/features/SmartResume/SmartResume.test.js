import React from 'react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import renderer from 'react-test-renderer';

import resumeReducer from './slicers/resumeSlicer';

import SmartResume from './SmartResume';

const store = configureStore({
  reducer: {
    resume: resumeReducer
  },
});

test('renders SmartResume', () => {
  const smartResume = renderer.create(
    <Provider store={store}>
      <SmartResume />
    </Provider>
  ).toJSON();

  expect(smartResume).toMatchSnapshot();
});
