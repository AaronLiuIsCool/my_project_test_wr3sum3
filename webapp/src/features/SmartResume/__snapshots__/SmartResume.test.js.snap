// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div>
  <div>
    智能创建
  </div>
  <div>
    Navigation on the left
  </div>
  <div>
    <p>
      基本信息
    </p>
    <p>
      教育背景
    </p>
    <div>
      <div
        className="experienceContainer"
      >
        <p>
          Kuaidao
        </p>
        <p>
          Intern
        </p>
      </div>
      <div
        className="experienceContainer"
      >
        <p>
          Kuaidao
        </p>
        <p>
          Software Engineer
        </p>
      </div>
    </div>
    <p>
      项目经历
    </p>
    <p>
      其他经历
    </p>
    <p>
      技能证书
    </p>
  </div>
  <div>
    Preview on right
  </div>
</div>
`;

exports[`renders SmartResume 1`] = `
<div>
  <div>
    智能创建
  </div>
  <div>
    Navigation on the left
  </div>
  <div>
    <p>
      基本信息
    </p>
    <p>
      教育背景
    </p>
    <div>
      <div
        className="experienceContainer"
      >
        <p>
          Kuaidao
        </p>
        <p>
          Intern
        </p>
      </div>
      <div
        className="experienceContainer"
      >
        <p>
          Kuaidao
        </p>
        <p>
          Software Engineer
        </p>
      </div>
    </div>
    <p>
      项目经历
    </p>
    <p>
      其他经历
    </p>
    <p>
      技能证书
    </p>
  </div>
  <div>
    Preview on right
  </div>
</div>
`;
